metadata-ingestion/src/datahub/ingestion/source/redash.py

```diff
@@ -552,9 +552,7 @@ def _get_dashboard_snapshot(self, dashboard_data):
             lastModified=AuditStamp(time=modified_ts, actor=modified_actor),
         )
 
-        dashboard_url = (
-            f"{self.config.connect_uri}/dashboard/{dashboard_data.get('slug', '')}"
-        )
+        dashboard_url = f"{self.config.connect_uri}/dashboards/{dashboard_data.get('id')}-{dashboard_data.get('slug', '')}"
```

Comment:
Is this format of the dashboard url documented in some public docs on the redash site? Trying to understand if accepting this might break other installations where this url looks different (maybe for older versions of redash etc.)

Reply:
Thanks for the review. I checked the public [docs](https://redash.io/help/user-guide/dashboards/dashboard-editing) on redash site. 
In this docs:

> Dashboard ids are guaranteed to be unique. But multiple dashboards may use the same name (and therefore `slug`). If a user visits `/dashboard/account-overview` and more than one dashboard exists with that slug, they will be redirected to the earliest created dashboard with that slug.

Since slug isn't guaranteed to be unique it should be:
if id is present dashboard url should be `{uri}/dashboards/<id>`  and if id is not present then dashboard url should be `{uri}/dashboard/<slug>`

But the `{uri}/dashboard/<slug>` endpoint is supported from redash version `10.0.0-beta`([ref1](https://github.com/getredash/redash/blob/4186f8303ee28d5cde7ca248c57df9023eba6224/client/app/pages/dashboards/DashboardPage.jsx#L205-L220), [ref2](https://github.com/getredash/redash/blob/master/CHANGELOG.md#bug-fixes), [ref3](https://redash.io/help/user-guide/integrations-and-api/api)). 

Therefore, to maintain backwards compatibility, the workaround is to check the version and create the dashboard URL to match the version

I tested on redash versions 10, 9, 8 and 7.